topi/python/topi/x86/int8_intrinsics.py

```diff
@@ -0,0 +1,107 @@
+"""Core kernel of dot product of 4 Int8 operations"""
+#pylint: disable=invalid-name
+import tvm
+
+
+def _intrin_reduce4int8_common(vec_size, num_elements_intel):
+    data = tvm.placeholder((num_elements_intel,), dtype='uint8', name='data')
+    kernel = tvm.placeholder((vec_size, num_elements_intel), dtype='int8', name='kernel')
+    k = tvm.reduce_axis((0, num_elements_intel), name='k')
+    C = tvm.compute((vec_size,),
+                    lambda i: tvm.sum(data[k].astype('int32') *
+                                      kernel[i, k].astype('int32'),
+                                      axis=k),
+                    name="C")
+
+    a_buffer = tvm.decl_buffer(data.shape, dtype='uint8', name="a_buffer",
+                               offset_factor=1,
+                               strides=[1])
+    b_buffer = tvm.decl_buffer(kernel.shape, dtype='int8', name="b_buffer",
+                               offset_factor=1,
+                               strides=[tvm.var('ldw'), 1])
```

Comment:
I feel all these strides bindings are unnecessary and can be removed.

Reply:
That's bizarre. In my understanding, the strides is implicitly inferred (given input tensor is compact), and `var(ldw)` is for binding the inferred strides. Actually if you changed innermost stride `1` to some other number, I expect it would fail with some binding mismatch error.
@tqchen Could you help with this? I'm also not fully understand the strides for buffer.

Reply:
#1725 The usage here is correct, thus it does not block merging this PR.